src/messages/manifestjson.js

```diff
@@ -35,6 +35,16 @@ export const MANIFEST_FIELD_INVALID = {
   file: MANIFEST_JSON,
 };
 
+export const MANIFEST_BAD_PERMISSION = {
+  code: 'MANIFEST_BAD_PERMISSION',
+  legacyCode: null,
+  message: _('The permission type is unsupported.'),
+  description: _(singleLineString`Permissions must be strings; object
```

Comment:
Don’t know of anything else yet, and it otherwise passed validation okay.
I’d like to get this merged into today’s tag, but if anything else comes up
I’ll look for a more definitive way to flag something as an app.